### PR TITLE
fix(replication): azurecr replication with token (#16888)

### PR DESCRIPTION
**Pull request by chlins**
Fix azurecr use ACR token failed to list tags, the root cause is the
scope action of acr token is 'metadata_read' not 'pull' when list v2
tags API.

Signed-off-by: chlins <chenyuzh@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes [#(16888)](https://github.com/goharbor/harbor/issues/16888)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/pkg/reg/adapter/azurecr/auth.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/docker/distribution/registry/client/auth/challenge"
+	commonhttp "github.com/goharbor/harbor/src/common/http"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/pkg/reg/model"
+	"github.com/goharbor/harbor/src/pkg/registry/auth"
+)
+
+var (
+	tagListRegexp = regexp.MustCompile("/v2/(" + reference.NameRegexp.String() + ")/tags/list")
```
Comment (stonezdj) [inline src/pkg/reg/adapter/azurecr/auth.go:35]: use https://github.com/goharbor/harbor/blob/b1580866427dfef90356a429164f29ecd0cb897a/src/lib/patterns.go#L24-L23 instead

File: src/pkg/reg/adapter/azurecr/auth.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	if tk.AccessToken != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", tk.AccessToken))
+	}
+
+	return nil
+}
+
+// buildTokenAPI builds token request API path.
+func (a *authorizer) buildTokenAPI(u *url.URL) (*url.URL, error) {
+	url, err := url.Parse(u.Scheme + "://" + u.Host + "/v2/")
```
Comment (stonezdj) [inline src/pkg/reg/adapter/azurecr/auth.go:116]: Rename url to avoid confuse

File: src/pkg/reg/adapter/azurecr/auth.go (hunk 1 above)
Comment (wy65701436) [reply src/pkg/reg/adapter/azurecr/auth.go:35]: +1

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/reg/adapter/azurecr/auth.go (hunk 1 above)
Comment (chlins) [reply src/pkg/reg/adapter/azurecr/auth.go]: updated

File: src/pkg/reg/adapter/azurecr/auth.go (hunk 2 above)
Comment (chlins) [reply src/pkg/reg/adapter/azurecr/auth.go]: updated

**Review by stonezdj — APPROVED** [review]

**Pull request merged**